presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java

```diff
@@ -354,6 +354,21 @@ public synchronized boolean isSubmissionFinished()
             return dispatchManager.waitForDispatched(queryId);
         }
 
+        /**
+         * Verifies the token of the query and returns the result.
+         * <p>
+         * Checks whether the token is the last one or the next one, comparing the value
+         * of last token and the value expected.
+         * At the end, results of the query is returned. Containing the token, the uriInfo
+         * containing the metadata and the query statistics.
+         *
+         * @param token generated token.
+         * @param uriInfo an object containing the uri metadata.
+         *
+         * @return the next token, the uri metadata and the informations about the query.
```

Comment:
Is this correct?
In my opinion this method is returning only a QueryResults object (you may link to this class)